server/src/main/java/org/elasticsearch/common/io/stream/NotSerializableExceptionWrapper.java

```diff
@@ -36,7 +37,7 @@
 public final class NotSerializableExceptionWrapper extends ElasticsearchException {
 
     private final String name;
-    private final RestStatus status;
+    private final HttpErrorStatus status;
```

Comment:
I wonder if we even need this. Looking at the usages of `status()` there's one use in `DefaultShardOperationFailedException` which this PR removes and another in `SQLExceptions.status` - which all look like they could be removed too. 

Maybe https://github.com/crate/crate/blob/f3ff8fe20edc47a7b452e33da68e96a718213636/server/src/main/java/org/elasticsearch/index/shard/IndexShard.java#L1450 is a bit trickier to replace(?)

Reply:
Yes, at `master`, the default `ElasticsearchException.httpErrorStatus()` impl.  (`UNHANDLED_SERVER_ERROR`) is used for the `NotSerializableException`. But given that we want to move to `HttpErrorStatus` in general everywhere, I'm not sure I understand your point correctly about that this isn't needed. Can you please clarify?

Reply:
> But given that we want to move to HttpErrorStatus in general everywhere, I'm not sure I understand your point correctly about that this isn't needed. Can you please clarify?

My thinking was that if the exception has a specific http status code we should probably make it an `ElasticsearchException` that's streamable, so there wouldn't be a need for a `status` field in `NotSerializableExceptionWrapper` because it would always end up in the default `return RestStatus.INTERNAL_SERVER_ERROR;` case of `SQLExceptions.status()` anyway